### ISSUE TITLE
[build] caching on Ubuntu builders of GH actions does nothing

### ISSUE THREAD

**Issue by lzutao** [rust-lang/rust-clippy#6155]
See this build log: <https://github.com/rust-lang/rust-clippy/pull/6154/checks?check_run_id=1235351421#step:6:1>

A solution is to cache only need files and subdirs: <https://doc.rust-lang.org/nightly/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci>.
We might also want to cache `.crates.toml` if we use `cargo install`.

**Comment by flip1995** [comment]
Yes, we're waiting for https://github.com/actions-rs/install/pull/5 to get through, so we can do this more cleanly.
